src/Core/Notifications/IDisabledNotificationsStorage.cs

```diff
@@ -32,15 +42,101 @@ public interface IDisabledNotificationsStorage
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class DisabledNotificationsStorage : IDisabledNotificationsStorage
     {
+        private readonly IVsVersionProvider vsVersionProvider;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        private string filePath => GetFilePath();
+
+        private DisabledNotifications disabledNotifications = null;
+        private readonly object lockObject = new object();
+
+        [ImportingConstructor]
+        public DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger) : this(vsVersionProvider, logger, new FileSystem())
+        {
+        }
+
+        internal /*for testing*/ DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger, IFileSystem fileSystem)
+        {
+            this.vsVersionProvider = vsVersionProvider;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
         public void DisableNotification(string id)
         {
-            // todo
+            lock (lockObject)
+            {
+                if (IsNotificationDisabled(id))
+                {
+                    return;
+                }
+                if (disabledNotifications == null)
+                {
+                    return;
+                }
+
+                disabledNotifications.AddNotification(id);
+                SaveNotifications();
+            }
         }
 
         public bool IsNotificationDisabled(string id)
         {
-            // todo
-            return false;
+            lock (lockObject)
+            {
+                if (disabledNotifications == null)
+                {
+                    disabledNotifications = ReadDisabledNotifications();
+                }
+                if (disabledNotifications == null)
+                {
+                    logger.WriteLine(Strings.DisabledNotificationsFailedToLoad);
+                    return false;
+                }
+                return disabledNotifications.Notifications.Any(n => n.Id == id);
+            }
+        }
+
+        private DisabledNotifications ReadDisabledNotifications()
+        {
+            try
+            {
+                if (!fileSystem.File.Exists(filePath)) 
+                { 
+                    logger.LogDebug($"[Notifications]Disabled notifications file does not exist. File: {filePath}"); 
+                    return new DisabledNotifications(); 
+                }
+
+                var fileContent = fileSystem.File.ReadAllText(filePath);
+                return JsonConvert.DeserializeObject<DisabledNotifications>(fileContent);
+            }
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+            {
+                logger.WriteLine(string.Format(Strings.DisabledNotificationReadError, ex.Message));                
+            }
+            return null;
```

Comment:
If the json is corrupted, we will forever return null and then never update any storage. 
I think this method should never return null, and worst case it should treat it as if the file does not exist. @duncanp-sonar wdyt?